### PR TITLE
Fix the checks of some functions

### PR DESCRIPTION
**Pull request by PeiweiHu**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by slontis** [comment]
The failing test requires the incorrect test to change from "RSA" to "RSA-PSS"
````
static int test_EVP_rsa_pss_with_keygen_bits(void)
{
    ....
    ....
    ret = TEST_ptr(md)
        && TEST_ptr((ctx = EVP_PKEY_CTX_new_from_name(testctx, "RSA-PSS", testpropq)))
````

RSA-PSS is a separate type.. It was just ignoring the error before your change

**Review by t8m — CHANGES_REQUESTED** [review]

**Review by t8m — APPROVED** [review]

File: crypto/rsa/rsa_lib.c [hunk 1]
```diff
@@ -1100,8 +1101,9 @@ int EVP_PKEY_CTX_set0_rsa_oaep_label(EVP_PKEY_CTX *ctx, void *label, int llen)
                                              (void *)label, (size_t)llen);
     *p++ = OSSL_PARAM_construct_end();
 
-    if (!evp_pkey_ctx_set_params_strict(ctx, rsa_params))
-        return 0;
+    ret = evp_pkey_ctx_set_params_strict(ctx, rsa_params);
+    if (ret <= 0)
+        return ret;
```
Comment (slontis) [inline crypto/rsa/rsa_lib.c:1106]: this will leak label..

Comment (t8m) [reply crypto/rsa/rsa_lib.c:1106]: Nope, the ownership of the label from caller is not transferred on failure and that is perfectly correct.

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master and 3.0 branches. Thank you for your contribution.

**Pull request closed without merging**